src/www/widgets/widgets/interface_list.widget.php

```diff
@@ -112,61 +145,27 @@ function interface_widget_update(sender, data)
+<table class="table table-striped table-condensed">
 <?php
-  $ifsinfo = get_interfaces_info();
-  foreach ($interfaces as $ifdescr => $ifname):
-    $listed = in_array($ifdescr, $pconfig['interfaceslistfilter']);
+  foreach ($interfaces as $ident => $ifname):
+    $listed = in_array($ident, $pconfig['interfaceslistfilter']);
     $listed = !empty($pconfig['interfaceslistinvert']) ? $listed : !$listed;
     if (!$listed) {
       continue;
     }
-    $ifinfo = $ifsinfo[$ifdescr]; ?>
-  <tr id="interface_widget_item_<?= html_safe($ifname) ?>">
-    <td style="width:5%; word-break: break-word;">
-<?php if (isset($ifinfo['ppplink'])): ?>
-      <span title="3g" class="fa fa-mobile text-success"></span>
-<?php elseif (isset($config['interfaces'][$ifdescr]['wireless'])): ?>
-<?php if ($ifinfo['status'] == 'associated' || $ifinfo['status'] == 'up'): ?>
-      <span title="wlan" class="fa fa-signal text-success"></span>
-<?php else: ?>
-      <span title="wlan_d" class="fa fa-signal text-danger"></span>
-<?php endif ?>
-<?php else: ?>
-<?php if ($ifinfo['status'] == 'up'): ?>
-      <span title="cablenic" class="fa fa-exchange text-success"></span>
-<?php else: ?>
-      <span title="cablenic" class="fa fa-exchange text-danger"></span>
-<?php endif ?>
-<?php endif ?>
-    </td>
-    <td style="width:15%; word-break: break-word;">
-      <strong>
-        <u>
-          <span onclick="location.href='/interfaces.php?if=<?= html_safe($ifdescr) ?>'" style="cursor:pointer">
-            <?= html_safe($ifname) ?>
-          </span>
-        </u>
-      </strong>
-    </td>
-    <td style="width:5%; word-break: break-word;">
-<?php if ($ifinfo['status'] == 'up' || $ifinfo['status'] == 'associated'): ?>
-      <span class="fa fa-arrow-up text-success"></span>
-<?php elseif ($ifinfo['status'] == "down"): ?>
-      <span class="fa fa-arrow-down text-danger"></span>
-<?php elseif ($ifinfo['status'] == "no carrier"): ?>
-      <span class="fa fa-times text-danger"></span>
-<?php else: ?>
-      <?= html_safe($ifinfo['status']) ?>
-<?php endif ?>
-    </td>
-    <td style="width:32%; word-break: break-word;">
-      <?= html_safe(empty($ifinfo['media']) ? $ifinfo['cell_mode'] ?? '' : $ifinfo['media']) ?>
-    </td>
+
+    list($primary4,, $bits4) = interfaces_primary_address($ident);
+    list($primary6,, $bits6) = interfaces_primary_address6($ident);
+?>
+  <tr id="interface_widget_item_<?= html_safe($ident) ?>">
+    <td style="width:5%; word-break: break-word;" id="link_status"></td>
+    <td style="width:15%; word-break: break-word;" id="if_name"></td>
+    <td style="width:5%; word-break: break-word;" id="if_status"></td>
+    <td style="width:32%; word-break: break-word;" id="if_media"></td>
     <td style="width:43%; word-break: break-word;">
-      <?= html_safe($ifinfo['ipaddr']) ?>
-      <?= !empty($ifinfo['ipaddr']) ? '<br/>' : '' ?>
-      <?= html_safe(interfaces_has_prefix_only($ifdescr) ? $ifinfo['linklocal'] : $ifinfo['ipaddrv6']) ?>
```

Comment:
@fichtner this one is annoying, any particular opinion about it?

Reply:
interfaces_primary_address6() should do exactly this to be honest